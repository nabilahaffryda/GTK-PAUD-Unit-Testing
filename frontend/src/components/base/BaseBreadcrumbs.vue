<template>
  <v-breadcrumbs large :items="items" divider="/">
    <template v-if="useIcon" v-slot:divider>
      <v-icon>{{ icon }}</v-icon>
    </template>
    <template v-slot:item="{ item }">
      <v-breadcrumbs-item :class="[{ link: item.to }, { 'secondary--text': item.to }]">
        <div v-if="item.to" @click="toLink(item)" style="cursor: pointer">
          {{ item.text }}
        </div>
        <div v-else style="cursor: default">{{ item.text }}</div>
      </v-breadcrumbs-item>
    </template>
  </v-breadcrumbs>
</template>

<script>
export default {
  props: {
    items: {
      type: Array,
      default: () => [],
    },
    icon: {
      type: String,
      default: 'mdi-chevron-right',
    },
    useIcon: {
      type: Boolean,
      default: true,
    },
  },
  methods: {
    toLink(item) {
      this.$router.push({
        name: item.to,
        params: Object.assign({}, item.params),
      });
    },
  },
};
</script>
