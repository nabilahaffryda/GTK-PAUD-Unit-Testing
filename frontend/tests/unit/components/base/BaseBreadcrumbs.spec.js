import { mount, createLocalVue } from "@vue/test-utils";
import BaseBreadcrumbs from "@/components/base/BaseBreadcrumbs.vue";
import Vuetify from 'vuetify';
import Vue from 'vue';
import VueRouter from 'vue-router';

Vue.use(Vuetify)
const localVue = createLocalVue();
localVue.use(VueRouter)
const router = new VueRouter();

describe('BaseBreadcrumbs.vue', () => {
    let vuetify
    beforeEach(() => {
        vuetify = new Vuetify()
    })
    function wrapperFactory({ } = {}) {
        return mount(BaseBreadcrumbs, {
            localVue,
            vuetify,
            router,
        });
    }
    test('call toLink method', () => {
        const wrapper = wrapperFactory();
        wrapper.vm.toLink = jest.fn();
        wrapper.vm.toLink();
        expect(wrapper.vm.toLink.mock.calls.length).toBe(1);
    })
})
