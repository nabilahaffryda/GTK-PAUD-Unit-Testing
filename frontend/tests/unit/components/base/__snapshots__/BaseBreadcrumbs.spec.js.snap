// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseBreadcrumbs.vue should show breadcrumbs 1`] = `<div large="" divider="/"></div>`;
